### DISCUSSION TITLE
 DJL v0.17.0 release

### DISCUSSION THREAD

**Discussion by zachgk** [deepjavalibrary/djl#1643] [News]
## Key Features

* Adds linux AArch64 support for PyTorch
* Upgrades Engine Releases
    * XGBoost version 1.6.0 ([#1624](https://github.com/deepjavalibrary/djl/pull/1624))
    * PaddlePaddle version 2.2.2 ([#1601](https://github.com/deepjavalibrary/djl/pull/1601))
    * ONNXRuntime version 1.11.0 ([#1602](https://github.com/deepjavalibrary/djl/pull/1602))
    * PyTorch version 1.11.0 ([#1583](https://github.com/deepjavalibrary/djl/pull/1583))
    * Apache MXNet version 1.9.0 ([#1429](https://github.com/deepjavalibrary/djl/pull/1429))
* Newly Added Datasets
    * PennTreebank dataset ([#1580](https://github.com/deepjavalibrary/djl/pull/1580))
    * WikiText-2 dataset ([#1545](https://github.com/deepjavalibrary/djl/pull/1545))
* Support freezing parameters for transfer learning ([#1544](https://github.com/deepjavalibrary/djl/pull/1544))

## Enhancement

* Upgrade djl-bench to 0.15.0 ([#1476](https://github.com/deepjavalibrary/djl/pull/1476))
* Add SessionOptions support for OnnxRuntime ([#1479](https://github.com/deepjavalibrary/djl/pull/1479))
* Device Name parsing ([#1490](https://github.com/deepjavalibrary/djl/pull/1490))
* arena allocator setter ([#1510](https://github.com/deepjavalibrary/djl/pull/1510))
* Upgrade native build mxnet version ([#1517](https://github.com/deepjavalibrary/djl/pull/1517))
* Refactor ml extension support ([#1521](https://github.com/deepjavalibrary/djl/pull/1521))
* Parse default device with Device.fromName ([#1529](https://github.com/deepjavalibrary/djl/pull/1529))
* Upgrade dependency versions ([#1533](https://github.com/deepjavalibrary/djl/pull/1533))
* release Update document version to 0.16.0 ([#1536](https://github.com/deepjavalibrary/djl/pull/1536))
* Bump up version to 0.17.0 ([#1537](https://github.com/deepjavalibrary/djl/pull/1537))
* Building DJL For aarch64 ([#1526](https://github.com/deepjavalibrary/djl/pull/1526))
* api Make resource loading compatible with java9 module ([#1541](https://github.com/deepjavalibrary/djl/pull/1541))
* pytorch Allows multiple native jars to package into fat jar ([#1543](https://github.com/deepjavalibrary/djl/pull/1543))
* Add document with installation instructions ([#1548](https://github.com/deepjavalibrary/djl/pull/1548))
* Separate AbstractSymbolBlock from AbstractBlock ([#1555](https://github.com/deepjavalibrary/djl/pull/1555))
* Add better error message for libstdc++ tf errors ([#1570](https://github.com/deepjavalibrary/djl/pull/1570))
* basicdataset Add Stanford Question Answering Dataset ([#1554](https://github.com/deepjavalibrary/djl/pull/1554))
* xgb Set default missing value to NaN ([#1571](https://github.com/deepjavalibrary/djl/pull/1571))
* allow empty batch data ([#1569](https://github.com/deepjavalibrary/djl/pull/1569))
* pytorch Allows load libtroch from pip installation package ([#1577](https://github.com/deepjavalibrary/djl/pull/1577))
* HF Tokenizer: get charspans ([#1584](https://github.com/deepjavalibrary/djl/pull/1584))
* benchmark Allows benchmark run on aarch64 for PyTorch ([#1591](https://github.com/deepjavalibrary/djl/pull/1591))
* add pytorch cuDNN acceleration ([#1592](https://github.com/deepjavalibrary/djl/pull/1592))
* add troubleshoot issue ([#1600](https://github.com/deepjavalibrary/djl/pull/1600))
* add paddle CU110 ([#1604](https://github.com/deepjavalibrary/djl/pull/1604))
* Update badge ([#1610](https://github.com/deepjavalibrary/djl/pull/1610))
* ONNXRuntime add tensorRT option ([#1611](https://github.com/deepjavalibrary/djl/pull/1611))
* api Refactor metrics API ([#1613](https://github.com/deepjavalibrary/djl/pull/1613))
* api Fixes metric dimension ([#1614](https://github.com/deepjavalibrary/djl/pull/1614))
* bom Update bom dependency ([#1615](https://github.com/deepjavalibrary/djl/pull/1615))
* pytorch Use precxx11 build for aarch64 native library ([#1619](https://github.com/deepjavalibrary/djl/pull/1619))
* pytorch use precxx11 for aarch64 ([#1620](https://github.com/deepjavalibrary/djl/pull/1620))
* Update inference_performance_optimization.md ([#1621](https://github.com/deepjavalibrary/djl/pull/1621))
* Fix testFreezeParameters for multi-gpu ([#1623](https://github.com/deepjavalibrary/djl/pull/1623))
* Support gather of pytorch ([#1622](https://github.com/deepjavalibrary/djl/pull/1622))

## Documentation and Examples

* Update README for 0.15.0 release ([#1477](https://github.com/deepjavalibrary/djl/pull/1477))
* Update README to use 0.16.0-SNAPSHOT version ([#1486](https://github.com/deepjavalibrary/djl/pull/1486))
* typo fix ([#1519](https://github.com/deepjavalibrary/djl/pull/1519))
* docs Fixes dataset document. ([#1523](https://github.com/deepjavalibrary/djl/pull/1523))
* jupyter remove unecessary maven import ([#1540](https://github.com/deepjavalibrary/djl/pull/1540))
* examples Update maven pom file dependency version ([#1546](https://github.com/deepjavalibrary/djl/pull/1546))
* Add release note to README ([#1565](https://github.com/deepjavalibrary/djl/pull/1565))
* update pytorch build instruction on android ([#1630](https://github.com/deepjavalibrary/djl/pull/1630))
* bump up versioning ([#1633](https://github.com/deepjavalibrary/djl/pull/1633))
* update android package versions ([#1635](https://github.com/deepjavalibrary/djl/pull/1635))
* docs Update pytorch and paddle version ([#1634](https://github.com/deepjavalibrary/djl/pull/1634))

## Breaking Changes

* Custom symbol blocks should extend `AbstractSymbolBlock` instead of `AbstractBlock`

## Bug Fixes

* Fix flaky test for tensorflow 2.7.0 on GPU ([#1475](https://github.com/deepjavalibrary/djl/pull/1475))
* Fixes topK items for DetectedObjects and make it configurable to Classifications ([#1478](https://github.com/deepjavalibrary/djl/pull/1478))
* Fix load native library failure on Android ([#1485](https://github.com/deepjavalibrary/djl/pull/1485))
* Adding huggingface tokenizer extension to BOM ([#1487](https://github.com/deepjavalibrary/djl/pull/1487))
* Fixes [#1149](https://github.com/deepjavalibrary/djl/pull/1149), fix NPE bug ([#1492](https://github.com/deepjavalibrary/djl/pull/1492))
* Fix djl site vue version ([#1495](https://github.com/deepjavalibrary/djl/pull/1495))
* Fixes MLP example code in README ([#1497](https://github.com/deepjavalibrary/djl/pull/1497))
* Fixes jni dependency in README document ([#1513](https://github.com/deepjavalibrary/djl/pull/1513))
* Fixes memory leak in hybrid engine ([#1518](https://github.com/deepjavalibrary/djl/pull/1518))
* Fix the version issue reported from get-pip.py ([#1530](https://github.com/deepjavalibrary/djl/pull/1530))
* Fix FastText JNI build ([#1531](https://github.com/deepjavalibrary/djl/pull/1531))
* api Fixes loading BlockFactory bug ([#1547](https://github.com/deepjavalibrary/djl/pull/1547))
* tensorflow Fixes tensorflow session always on gpu(0) bug ([#1558](https://github.com/deepjavalibrary/djl/pull/1558))
* xgb Fixes missing anonymous classes ([#1572](https://github.com/deepjavalibrary/djl/pull/1572))
* examples Fix ImageClassification invalid probability ([#1575](https://github.com/deepjavalibrary/djl/pull/1575))
* fix naming ([#1581](https://github.com/deepjavalibrary/djl/pull/1581))
* ONNXRuntime fix naming ([#1608](https://github.com/deepjavalibrary/djl/pull/1608))
* basicdataset Fixed out of bound limit ([#1599](https://github.com/deepjavalibrary/djl/pull/1599))
* api Avoid NPE in Metric.toString() ([#1626](https://github.com/deepjavalibrary/djl/pull/1626))
* integration Enable gather unit test for windows ([#1638](https://github.com/deepjavalibrary/djl/pull/1638))
* Add rpath fix to Native Publish PyTorch ([#1639](https://github.com/deepjavalibrary/djl/pull/1639))
* api Fixes JDK 18 compiler warnings ([#1640](https://github.com/deepjavalibrary/djl/pull/1640))

## Contributors

* @AKAGIwyf
* @andreabrduque
* @dandansamax
* @frankfliu
* @hd1080p
* @KexinFeng
* @lanking520
* @patins1
* @sindhuvahinis
* @WHALEEYE
* @zachgk

## New Contributors

* @sindhuvahinis made their first contribution in https://github.com/deepjavalibrary/djl/pull/1519
* @KexinFeng made their first contribution in https://github.com/deepjavalibrary/djl/pull/1530
* @hd1080p made their first contribution in https://github.com/deepjavalibrary/djl/pull/1526
* @dandansamax made their first contribution in https://github.com/deepjavalibrary/djl/pull/1545
* @WHALEEYE made their first contribution in https://github.com/deepjavalibrary/djl/pull/1554
* @patins1 made their first contribution in https://github.com/deepjavalibrary/djl/pull/1569
* @AKAGIwyf made their first contribution in https://github.com/deepjavalibrary/djl/pull/1580

**Full Changelog**: https://github.com/deepjavalibrary/djl/compare/750c153a7...v0.17.0

<hr /><em>This discussion was created from the release <a href='https://github.com/deepjavalibrary/djl/releases/tag/v0.17.0'> DJL v0.17.0 release</a>.</em>
